src/python/gudhi/simplex_tree.pyx

```diff
@@ -38,13 +38,27 @@ cdef class SimplexTree:
     cdef Simplex_tree_persistence_interface * pcohptr
 
     # Fake constructor that does nothing but documenting the constructor
-    def __init__(self):
+    def __init__(self, other = None):
         """SimplexTree constructor.
+
+        :param other: If `other` is `None` (default value), an empty `SimplexTree` is created.
+            If `other` is a `SimplexTree`, the `SimplexTree` is constructed from a deep copy of `other`.
+        :type other: SimplexTree (Optional)
+        :returns: An empty or a copy simplex tree.
+        :rtype: SimplexTree
+
+        :note: If the `SimplexTree` is a copy, it requires :func:`compute_persistence` to be launched again as the
+            persistence result is not copied.
         """
 
     # The real cython constructor
-    def __cinit__(self):
-        self.thisptr = <intptr_t>(new Simplex_tree_interface_full_featured())
+    def __cinit__(self, other = None):
+        cdef SimplexTree ostr
+        if other and type(other) is SimplexTree:
+            ostr = <SimplexTree> other
+            self.thisptr = <intptr_t>(new Simplex_tree_interface_full_featured(dereference(ostr.get_ptr())))
+        else:
+            self.thisptr = <intptr_t>(new Simplex_tree_interface_full_featured())
```

Comment:
If I try to construct a SimplexTree from something random, say a string, it will construct an empty SimplexTree without complaining?

Reply:
You are right that it can lead to misunderstanding like `SimplexTree([1,2,3])`, ...
I moodified it to raise an exception on d5ac245